### DISCUSSION TITLE
Render Distant Zones with Low-Poly / View Distance Control

### DISCUSSION THREAD

**Discussion by Naros** [spimort/TerraBrush#80] [Q&A]
In large world games, terrains often use a similar chunk-based system that you have chosen for your zones.  However, these games often use a view distance to control how far into the distance the terrain is rendered while also having a secondary distance that controls whether the terrain is represented by some low-poly mesh that isn't splatted and takes on a hue of the distant atmosphere for performance reasons.

It would be neat if TerraBrush provided a way to support such zone features.

**Comment by spimort** [comment]
Hey @Naros , thanks for the feedback 🙂!

The way I handle it on my side is using Clipmap. I recently added the support for LOD on the terrain, using a Clipmap, which allows me to reduce the number of vertices for the terrain when the player is far (something like this : 
![image](https://github.com/spimort/TerraBrush/assets/6035485/7c9a9d50-86b5-4464-9360-43351ef123c4)
). Dependending on the LOD settings, if the Clipmap reaches the end, the terrain start to disappear.

Is this what you were looking for?
I also talk about it here : https://youtu.be/BardvKC0HF0?si=ELiefrFHUbQyEFcC

Also, for packedscenes, I try to rely on Godot to handle the visibility so a visibility range can be set on the object itself : 
![image](https://github.com/spimort/TerraBrush/assets/6035485/4b9625da-ba7a-4a64-b069-fb0e1e094ca6)

Please, I'd like to know if it covers what you were talking about 🙂!

**Comment by Naros** [comment]
Somewhat.  

What I was curious is whether you've thought about having the clipmap set to something reasonable for the hardware, lets say 500m and then beyond that to some other distance, lets say 750m or 1000m, you have a very low poly count pre-baked model that takes the place of the tile that doesn't follow clipmap rules but rather is specifically designed to provide horizon mountain effects that aren't shaded but gives the illusion of a farther render distance.

**Comment by spimort** [comment]
@Naros 
I see what you mean! 
The clipmap is already pretty configurable so you can do what ever you want with it but I did not think about "beyond" the clipmap.

That could be a cool improvement 🙂

**Comment by spimort** [comment]
I'll convert this issue to a discussion since it's more a question than a request.
